### PR TITLE
Maxpool fix bug

### PR DESCRIPTION
**Pull request by pszmel**
1. Add support for VALID pad_type
2. Add validation checks for proper attributes sizes
3. Add type_prop tests
4. Small change in the spec

ticket: 45334

### CODE REVIEW

**Comment by GlebKazantaev** [comment]
Please add ticket number to the PR description.

File: ngraph/core/src/op/max_pool.cpp [hunk 1]
```diff
@@ -99,23 +99,61 @@ void op::v1::MaxPool::validate_and_infer_types()
...
     }
 
-    const bool update_auto_padding_succeed =
-        update_auto_padding(arg_shape, m_pads_end, m_pads_begin);
-
+    bool update_auto_padding_succeed = true;
+    if (m_auto_pad == PadType::SAME_UPPER || m_auto_pad == PadType::SAME_LOWER)
+    {
+        CoordinateDiff pads_end;
+        CoordinateDiff pads_begin;
```
Comment (lazarevevgeny) [inline ngraph/core/src/op/max_pool.cpp:149]: Why do you need these variables?

**Review by lazarevevgeny — APPROVED** [review]
The overall looks good except the redundant non-used variables.

File: ngraph/core/src/op/max_pool.cpp (hunk 1 above)
Comment (pszmel) [reply ngraph/core/src/op/max_pool.cpp]: They are not needed, my bad.

**Review by mbencer — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**